LEGO1/lego/legoomni/src/entity/legojetskiraceactor.cpp

```diff
@@ -11,17 +22,120 @@ LegoJetskiRaceActor::LegoJetskiRaceActor()
 	m_unk0x150 = 1.5f;
 }
 
-// STUB: LEGO1 0x10081120
+// FUNCTION: LEGO1 0x10081120
+// FUNCTION: BETA10 0x100ce19f
 MxS32 LegoJetskiRaceActor::VTable0x1c(LegoPathBoundary* p_boundary, LegoEdge* p_edge)
 {
-	// TODO
-	return 0;
+	// These are almost certainly not the correct names, but they produce the correct BETA10 stack
+	Mx3DPointFloat a;
+	Mx3DPointFloat bbb;
+	Mx3DPointFloat c;
+
+	// These names are verified by an assertion below
+	Vector3* v1 = NULL;
+	Vector3* v2 = NULL;
+
+	if (m_state == 1) {
+		if (m_destEdge == LegoPathController::GetControlEdgeA(13)) {
+			m_boundary = (LegoPathBoundary*) m_destEdge->OtherFace(LegoPathController::GetControlBoundaryA(13));
+		}
+		else if (m_destEdge == LegoPathController::GetControlEdgeA(15)) {
+			m_boundary = (LegoPathBoundary*) m_destEdge->OtherFace(LegoPathController::GetControlBoundaryA(15));
+		}
+
+		m_state = 0;
+		m_unk0x7c = 0;
+
+		if (m_userNavFlag) {
+			NavController()->SetLinearVel(m_worldSpeed);
+			return 0;
+		}
+		else {
+			return 1;
+		}
+	}
+	else {
+		if (p_edge == LegoPathController::GetControlEdgeA(12)) {
+			m_state = 1;
+
+			if (m_worldSpeed < g_eight) {
+				m_worldSpeed = g_eight;
+			}
+
+			m_destEdge = LegoPathController::GetControlEdgeA(13);
+			m_boundary = LegoPathController::GetControlBoundaryA(13);
+		}
+		else if (p_edge == LegoPathController::GetControlEdgeA(14)) {
+			m_state = 1;
+
+			if (m_worldSpeed < g_eight) {
+				m_worldSpeed = g_eight;
+			}
+
+			m_destEdge = LegoPathController::GetControlEdgeA(15);
+			m_boundary = LegoPathController::GetControlBoundaryA(15);
+		}
+
+		if (m_state == 1) {
+			if (m_userNavFlag) {
+				m_unk0xe4 = 0.5f;
+			}
+
+			v1 = m_destEdge->CCWVertex(*m_boundary);
+			v2 = m_destEdge->CWVertex(*m_boundary);
+			assert(v1 && v2);
+
+			a[0] = (*v1)[0] + ((*v2)[0] - (*v1)[0]) * m_unk0xe4;
```

Comment:
This looks like `LERP3` in `vec.h`

Reply:
This has had a negative effect on the stack match in BETA10, but the refactor definitely looks cleaner and LEGO1 is unaffected.